pkg/m3admin/namespace.go

```diff
@@ -0,0 +1,115 @@
+	_defaultFlushEnabled                                             = true
+	_defaultWritesToCommitLog                                        = true
+	_defaultCleanupEnabled                                           = true
+	_defaultSnapshotEnabled                                          = false
+	_defaultRepairEnabled                                            = false
+	_defaultRententionOptionRetentionPeriodNanos                     = int64(172800000000000)
+	_defaultRententionOptionBlockSizeNanos                           = int64(7200000000000)
+	_defaultRententionOptionBufferFutureNanos                        = int64(600000000000)
+	_defaultRententionOptionBufferPastNanos                          = int64(600000000000)
+	_defaultRententionOptionBlockDataExpiry                          = true
+	_defaultRententionOptionBlockDataExpiryAfterNotAccessPeriodNanos = int64(300000000000)
+	_defaultIndexOptionenabled                                       = true
+	_defaultIndexOptionBlockSizeNanos                                = int64(7200000000000)
+)
+
+//NamespaceCreate will create a namespace
+func (m *m3admin) NamespaceCreate(cluster *myspec.M3DBCluster) error {
+	namespaceCreateRequest := admin.NamespaceAddRequest{
+		Name: cluster.GetObjectMeta().GetName(),
+		Options: &namespace.NamespaceOptions{
+			BootstrapEnabled:  _defaultBootstrapEnabled,
+			FlushEnabled:      _defaultFlushEnabled,
+			WritesToCommitLog: _defaultWritesToCommitLog,
+			CleanupEnabled:    _defaultCleanupEnabled,
+			RepairEnabled:     _defaultRepairEnabled,
+			RetentionOptions: &namespace.RetentionOptions{
+				RetentionPeriodNanos:                     _defaultRententionOptionRetentionPeriodNanos,
+				BlockSizeNanos:                           _defaultRententionOptionBlockSizeNanos,
+				BufferFutureNanos:                        _defaultRententionOptionBufferFutureNanos,
+				BufferPastNanos:                          _defaultRententionOptionBufferPastNanos,
+				BlockDataExpiry:                          _defaultRententionOptionBlockDataExpiry,
+				BlockDataExpiryAfterNotAccessPeriodNanos: _defaultRententionOptionBlockDataExpiryAfterNotAccessPeriodNanos,
+			},
+			SnapshotEnabled: _defaultSnapshotEnabled,
+			IndexOptions: &namespace.IndexOptions{
+				Enabled:        _defaultIndexOptionenabled,
+				BlockSizeNanos: _defaultIndexOptionBlockSizeNanos,
+			},
+		},
+	}
+	data, err := json.Marshal(namespaceCreateRequest)
+	if err != nil {
+		return err
+	}
+	// TODO(PS) - Remove hardcoded references
+	url := fmt.Sprintf("http://%s.%s:7201%s", _m3coordinatorServiceName, cluster.GetNamespace(), _namespaceURI)
+	request, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+	request.Header.Add("Content-Type", "application/json")
+	if _, err := m.doHttpRequest(request); err != nil {
+		return err
+	}
+	logrus.WithField("namespace", namespaceCreateRequest.GetName()).Info("successfully created namespace")
```

Comment:
We're trying to move the entire stack to use `zap` instead of other loggers. Mind using it instead of logrus?